scripts/outer/kubectl-kubestellar-prep_for_syncer

```diff
@@ -110,68 +113,85 @@ fi
 
 set -e
 
-imw_space_config="${PWD}/temp-space-config/spaceprovider-default-${imw}"
-kubectl-kubestellar-get-config-for-space --space-name ${imw} --sm-core-config $SM_CONFIG --sm-context $SM_CONTEXT --output $imw_space_config
-mbwsname=$(KUBECONFIG=$imw_space_config kubectl "${kubectl_flags[@]}" get synctargets.edge.kubestellar.io "$stname" -o jsonpath="{.metadata.annotations['kcp\.io/cluster']}-mb-{.metadata.uid}")
+
+imw_kubeconfig=$(mktemp "${imw}.kubeconfig#XXXX")
+kubectl-kubestellar-space-get_kubeconfig "${sub_flags[@]}" "${kubectl_flags[@]}" "${imw}" "$imw_kubeconfig"
+mbsname=$(KUBECONFIG=$imw_kubeconfig kubectl get synctargets.edge.kubestellar.io "$stname" -o jsonpath="{.metadata.annotations['kcp\.io/cluster']}-mb-{.metadata.uid}")
 
 if ! [[ "$imw" =~ [a-z0-9].* ]]; then
-    echo "$0: imw ${espw@Q} is not valid" >&2
+    echo "$0: imw '${imw}' is not valid" >&2
     exit 1
 fi
 
 if ! [[ "$espw" =~ [a-z0-9].* ]]; then
-    echo "$0: espw ${espw@Q} is not valid" >&2
+    echo "$0: espw '${espw}' is not valid" >&2
     exit 1
 fi
 
-espw_space_config="${PWD}/temp-space-config/spaceprovider-default-${espw}"
-kubectl-kubestellar-get-config-for-space --space-name ${espw} --sm-core-config $SM_CONFIG --sm-context $SM_CONTEXT --output $espw_space_config
+espw_kubeconfig=$(mktemp espw.kubeconfig#XXXX)
+kubectl-kubestellar-space-get_kubeconfig "${sub_flags[@]}" "${kubectl_flags[@]}" ${espw} "$espw_kubeconfig"
 
-if ! cluster_ns_name="$(KUBECONFIG=$espw_space_config kubectl "${kubectl_flags[@]}" -n kubestellar get configmap kbmap-"$imw" -ojsonpath="{.metadata.labels['kubestellar\.io\/kube-bind-id']}")"; then
+if ! cluster_ns_name="$(KUBECONFIG=$espw_kubeconfig kubectl -n kubestellar get configmap kbmap-"$imw" -ojsonpath="{.metadata.labels['kubestellar\.io\/kube-bind-id']}")"; then
```

Comment:
I see we are mixing KUBECONFIG=  and --kubeconfig.  I suggest using the --kubeconfig for direct kubectl commands

Reply:
OTOH, consistently starting lines with `KUBECONFIG=<something> ` has a certain appeal.

What do you mean by _direct_ kubectl commands?  Do you mean ones that do not involve a kubectl plugin? If so, why should that matter to this question?